src/test/java/cn/RecipeAPI/RecipeApiApplicationTests.java

```diff
@@ -1,13 +1,267 @@
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
 
-@SpringBootTest
-class RecipeApiApplicationTests {
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Set;
 
-	@Test
-	void contextLoads() {
-	}
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
 
+@WebMvcTest(RecipeController.class)
+@ContextConfiguration(classes = RecipeApiApplication.class)
+@ActiveProfiles(profiles = "test")
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class RecipeApiApplicationTests {
+    // Create some test recipes
+    Review review = Review.builder().description("was just caramel").rating(3).username("idk").build();
+    Review review2 = Review.builder().description("was just egg").rating(4).username("idk").build();
+    Recipe recipe = Recipe.builder().name("test name").difficultyRating(1).minutesToMake(5)
+            .ingredients(Set.of(Ingredient.builder().name("spam").amount("1 can").build()))
+            .steps(Set.of(Step.builder().stepNumber(1).description("eat spam").build()))
+            .locationURI(new URI("http://localhost:8080/recipes/1"))
+            .reviews(Set.of(review))
+            .id(1L)
+            .build();
+    Recipe recipe2 = Recipe.builder().name("test name2").difficultyRating(2).minutesToMake(6)
+            .ingredients(Set.of(Ingredient.builder().name("egg").amount("1 egg").build()))
+            .steps(Set.of(Step.builder().stepNumber(1).description("crack egg").build()))
+            .locationURI(new URI("http://localhost:8080/recipes/2"))
+            .reviews(Set.of(review2))
+            .id(2L)
+            .build();
+    ArrayList<Recipe> recipes = new ArrayList<>(Arrays.asList(recipe, recipe2));
+    @Autowired
+    private MockMvc mockMvc;
+    @MockBean
+    private RecipeService recipeService;
+
+    public RecipeApiApplicationTests() throws URISyntaxException {
+    }
+
+    @Test
+    void contextLoads() {
+    }
+
+    @Order(1)
+    @Test
+    public void testGetRecipeByIdSuccessBehavior() throws Exception {
+        final long recipeId = 1;
+
+        // When getRecipeById is called with any long id, it should return a recipe
+        when(recipeService.getRecipeById(anyLong())).thenReturn(recipe);
```

Comment:
I would suggest to do mocking as specific as possible.
```suggestion
        when(recipeService.getRecipeById(1l)).thenReturn(recipe);
```
This way if of endpoint modifies the `id` or does two calls for some reason or anything else it would be caught only by the specific test.